### PR TITLE
Flip on leak detector and suppress or fix remaining leak traces

### PR DESCRIPTION
**Pull request by bdero**
Patch/suppress the remaining leaks and flip on the leak detector to start catching new leaks in CI.

The only unaccounted for leaks remaining are in the linux shell. It's taking some time for me to figure out the correct behavior for some of those, but figured it's best to get the leak detector on sooner than later so that we can start detecting new leaks as they come.

### CODE REVIEW

File: testing/lsan_suppressions.txt [hunk 1]
```diff
@@ -40,3 +40,25 @@ leak:flutter::Engine::Run(flutter::RunConfiguration)
...
+# The RefCounted DebugChecks tests intentionally fail to wrap and manage heap
+# allocated objects.
+leak:RefCountedTest_DebugChecks_Test::TestBody
+
+# Objects allocated by this mock API are intentionally not collected and are
+# instead marked as released in order to explicitly assert double free attempts
+# and allow some tests to inspect contents.
+leak:*flutter/shell/platform/linux/testing/mock_engine.cc
+
+# TODO(bdero): Fix FL leaks
```
Comment (zanderso) [inline testing/lsan_suppressions.txt:53]: Please file a new issue or make a note in an existing issue, and add a link to the issue to this TODO comment.

File: testing/lsan_suppressions.txt [hunk 2]
```diff
@@ -40,3 +40,25 @@ leak:flutter::Engine::Run(flutter::RunConfiguration)
...
+leak:*flutter/shell/platform/linux/fl_keyboard_manager_test.cc*
+leak:*flutter/shell/platform/linux/fl_key_channel_responder_test.cc*
+leak:*flutter/shell/platform/linux/fl_basic_message_channel_test.cc*
+leak:fl_message_codec_decode_message
+
+# Unfortunately, realloc calls originating from g_realloc effectively obscure
```
Comment (zanderso) [inline testing/lsan_suppressions.txt:59]: Please file an issue.

File: testing/lsan_suppressions.txt (hunk 1 above)
Comment (bdero) [reply testing/lsan_suppressions.txt]: Good idea, done: https://github.com/flutter/flutter/issues/90155

File: testing/lsan_suppressions.txt (hunk 2 above)
Comment (bdero) [reply testing/lsan_suppressions.txt:60]: Done: https://github.com/flutter/flutter/issues/90156

File: shell/platform/embedder/tests/embedder_unittests_gl.cc [hunk 3]
```diff
@@ -3537,7 +3537,9 @@ TEST_F(EmbedderTest, CreateInvalidBackingstoreOpenGLTexture) {
         backing_store_out->open_gl.texture.target = 0;
         backing_store_out->open_gl.texture.name = 0;
         backing_store_out->open_gl.texture.format = 0;
-        backing_store_out->open_gl.texture.user_data = new TestCollectOnce();
+        static TestCollectOnce collect_once_user_data;
```
[the remark this answers is not in the document]
Comment (bdero) [reply shell/platform/embedder/tests/embedder_unittests_gl.cc:3540]: No, 1 render target should be created during this test, but it also looks like they wouldn't be made in parallel if there were multiple either: https://github.com/flutter/engine/blob/be8467ea6113d0811a3fb7bb4b2925f5cc310e92/shell/platform/embedder/embedder_external_view_embedder.cc#L179-L186

File: shell/platform/embedder/tests/embedder_unittests_gl.cc [hunk 4]
```diff
@@ -3597,8 +3599,10 @@ TEST_F(EmbedderTest, CreateInvalidBackingstoreOpenGLFramebuffer) {
         backing_store_out->open_gl.type = kFlutterOpenGLTargetTypeFramebuffer;
         backing_store_out->open_gl.framebuffer.target = 0;
         backing_store_out->open_gl.framebuffer.name = 0;
+        static TestCollectOnce collect_once_user_data;
```
[the remark this answers is not in the document]
Comment (bdero) [reply shell/platform/embedder/tests/embedder_unittests_gl.cc:3602]: Should be safe in this test

**Review by zanderso — APPROVED** [review]

**Pull request merged**